### PR TITLE
 Component RatioLinkedAscertainment

### PR DESCRIPTION
**Pull request by cdc-mitzimorris**
implemented #852,  updated tutorial.

### CODE REVIEW

File: docs/tutorials/ascertainment.qmd [hunk 1]
```diff
@@ -21,126 +21,138 @@ jupyter:
...
+
+The baseline rates $\boldsymbol{\alpha}_0$ set the prior location.
+The covariance matrix $\boldsymbol{\Sigma}$ controls the scale of variation in each logit-transformed rate and how strongly the rates vary together under the prior.
+
+For two signals, positive prior covariance means that prior draws with a higher-than-expected rate for one signal tend to have a higher-than-expected rate for the other.
+Negative covariance means that they tend to move in opposite directions.
+Zero covariance means that the multivariate normal prior does not couple their deviations from the baseline rates.
+
+The covariance structure is supplied to `JointAscertainment`; the component does not estimate an unknown covariance matrix from the observation data.
+It describes prior dependence, not a causal relationship between the signals.
```
Comment (dylanhmorris) [inline docs/tutorials/ascertainment.qmd:147]: One could infer $\Sigma$ rather than hard coding it, no? Just pass in a sampled quantity. I think simpler to say that `JointAscertainment` RVs treat $\Sigma$ as fixed; to infer $\Sigma$ you'd need to sample it upstream.

File: docs/tutorials/ascertainment.qmd [hunk 2]
```diff
@@ -21,126 +21,138 @@ jupyter:
...
 ```
 
-Ascertainment is the probability that a latent infection appears in an observed signal, e.g., a hospitalization or visit to the emergency department.
-For hospital admissions this probability is often called an infection-hospitalization rate (IHR).
-For emergency department visits it might be called an infection-ED-visit rate (IEDR).
+## Overview
 
-PyRenew count observations accept an `ascertainment_rate_rv`.
-For simple models this can be any ordinary `RandomVariable`.
+An ascertainment rate is the probability that an infection results in an observed event.
```
Comment (dylanhmorris) [inline docs/tutorials/ascertainment.qmd]:
```suggestion
An ascertainment rate is the rate at which infections lead to observed events.
```

File: docs/tutorials/ascertainment.qmd [hunk 3]
```diff
@@ -21,126 +21,138 @@ jupyter:
...
+For hospital admissions, this probability is often called the infection-hospitalization rate (IHR).
```
Comment (dylanhmorris) [inline docs/tutorials/ascertainment.qmd]:
```suggestion
For hospital admissions, this is often called the infection-hospitalization rate (IHR).
```

File: docs/tutorials/ascertainment.qmd [hunk 4]
```diff
@@ -154,103 +166,83 @@ joint_ascertainment = JointAscertainment(
...
-    )
-    + p9.geom_abline(slope=1, intercept=jnp.log10(2), linetype="dashed", color="gray")
-    + p9.scale_y_log10()
-    + p9.scale_x_log10()
-    + theme_tutorial
-)
-```
+The ratio may be any positive value.
+Unlike `JointAscertainment`, this component does not automatically constrain the derived linked rate to the unit interval.
+Users must choose priors that make $\alpha_{\mathrm{base}}\rho > 1$ negligibly likely.
```
Comment (dylanhmorris) [inline docs/tutorials/ascertainment.qmd]: You can imagine certain observables with more than one ascertained event per infection, so I don't think all ascertainment rates need to be probabilities. I think we can say something like

```suggestion
If both rates should be $<1$ (for instance because both are probabilities), make sure to choose priors that make $\alpha_{\mathrm{base}}\rho > 1$ negligibly likely.
```

File: test/test_ascertainment.py [hunk 5]
```diff
@@ -288,6 +292,125 @@ def test_signal_accessor_requires_active_context(self):
...
+        assert ascertainment.signals == ("ed_visits", "hospital")
+        assert ascertainment.base_signal == "ed_visits"
+        assert ascertainment.linked_signal == "hospital"
+        assert ascertainment.base_rate_rv is base_rate_rv
+        assert ascertainment.ratio_rv is ratio_rv
+
+        base_accessor = ascertainment.for_signal("ed_visits")
+        linked_accessor = ascertainment.for_signal("hospital")
+        assert base_accessor.signal_name == "ed_visits"
+        assert linked_accessor.signal_name == "hospital"
```
Comment (dylanhmorris) [inline test/test_ascertainment.py:359]: also assert the expected type and the expected `ascertainment_name`?

File: test/test_ascertainment.py [hunk 6]
```diff
@@ -288,6 +292,125 @@ def test_signal_accessor_requires_active_context(self):
...
+        ascertainment = RatioLinkedAscertainment(
+            name="he_ascertainment",
+            base_signal="ed_visits",
+            linked_signal="hospital",
+            base_rate_rv=DeterministicVariable("iedr", jnp.array(0.2)),
+            ratio_rv=DeterministicVariable("ihr_rel_iedr", jnp.array(0.5)),
+        )
+
+        with numpyro.handlers.trace() as trace:
+            values = ascertainment.sample(unused_context_value=True)
```
Comment (dylanhmorris) [inline test/test_ascertainment.py]: why `unused_context_value=True`?

File: test/test_ascertainment.py [hunk 7]
```diff
@@ -288,6 +292,125 @@ def test_signal_accessor_requires_active_context(self):
...
+    def test_sample_returns_base_rate_and_scaled_linked_rate(self):
+        """Test that the linked rate is the base rate times the ratio."""
+        ascertainment = RatioLinkedAscertainment(
+            name="he_ascertainment",
+            base_signal="ed_visits",
+            linked_signal="hospital",
+            base_rate_rv=DeterministicVariable("iedr", jnp.array(0.2)),
+            ratio_rv=DeterministicVariable("ihr_rel_iedr", jnp.array(0.5)),
+        )
+
```
Comment (dylanhmorris) [inline test/test_ascertainment.py:377]: Does this not need a seed? If it does, should we be concerned that the tests are passing?

File: test/test_ascertainment.py [hunk 8]
```diff
@@ -288,6 +292,125 @@ def test_signal_accessor_requires_active_context(self):
...
+            values["ed_visits"],
+            trace["he_ascertainment_ed_visits"]["value"],
+        )
+        assert jnp.array_equal(
+            values["hospital"],
+            trace["he_ascertainment_hospital"]["value"],
+        )
+
+    def test_sample_calls_each_random_variable_once(self):
+        """Test that one base-rate site and one ratio site are sampled."""
```
Comment (dylanhmorris) [inline test/test_ascertainment.py]:
```suggestion
        """Test that the base-rate site and the ratio site are sampled and converted into user-exposed, signal specific deterministic sites"""
```

**Review by dylanhmorris — CHANGES_REQUESTED** [review]
Thanks @cdc-mitzimorris. A few minor changes requested.

File: docs/tutorials/ascertainment.qmd (hunk 1 above)
Comment (cdc-mitzimorris) [reply docs/tutorials/ascertainment.qmd:147]: good point. simplified.

File: test/test_ascertainment.py (hunk 5 above)
Comment (cdc-mitzimorris) [reply test/test_ascertainment.py:359]: added.

File: test/test_ascertainment.py (hunk 6 above)
Comment (cdc-mitzimorris) [reply test/test_ascertainment.py]: good catch.  not needed - removed.

File: test/test_ascertainment.py (hunk 7 above)
Comment (cdc-mitzimorris) [reply test/test_ascertainment.py:377]: these are deterministic values - i.e., fixed - this test is for the returned signal mapping.   the next test is for stochastic sites creation.

**Comment by cdc-mitzimorris** [comment]
> Thanks @cdc-mitzimorris. A few minor changes requested.

@dylanhmorris all requested changes made (and questions answered).

**Review by dylanhmorris — APPROVED** [review]

**Comment by dylanhmorris** [comment]
Thanks, @cdc-mitzimorris!
